circuit-retained/src/commonMain/kotlin/com/slack/circuit/retained/CollectRetained.kt

```diff
@@ -40,3 +40,32 @@ public fun <T : R, R> Flow<T>.collectAsRetainedState(
       collect { value = it }
     } else withContext(context) { collect { value = it } }
   }
+
+/**
+ * Produces and collects values from a [Flow] provided by the [producer] lambda and represents its
+ * latest value via retained [State]. Every time there would be a new value posted into the [Flow],
+ * the returned [State] will be updated causing recomposition of every [State.value] usage.
+ *
+ * This function safely produces and collects a Flow by using a single retained state with unified
+ * inputs. The Flow is produced within the retained state's producer block, ensuring proper scoping
+ * and preventing memory leaks from retaining Flow instances across configuration changes.
```

Comment:
nit before we merge: let's unpack this a bit more, because it says "preventing memory leaks from retaining Flow instances" but we _are_ retaining the produced Flow instance no?